### ISSUE TITLE
Replace react-vis with something else?

### ISSUE THREAD

**Issue by PhilippWendler** [sosy-lab/benchexec#570]
The library [react-vis](https://github.com/uber/react-vis/) that we currently use for drawing plots is [unmaintained](https://github.com/uber/react-vis/blob/master/DEPRECATED.md). The company behind seems to be working on a successor, but it is unclear when this will be available and how it will be.

Right now the library does what we need, but at some point we might need to replace it. Based on this [overview survey](https://github.com/sosy-lab/benchexec/files/4516862/libraries.pdf), the library [recharts](https://github.com/recharts/recharts) might be a suitable replacement.

**Comment by DennisSimon** [comment]
`recharts` looks very promising. The api looks easy to use and easily expandable.

**Comment by lachnerm** [comment]
It seems that there has been some development going on after the date of this issue. The deprecation message [was removed](https://github.com/uber/react-vis/commit/b54ab8ab243770e4e04ab04e1161713925c52008) at the end of May and some changes happened. However, [here](https://github.com/uber/react-vis/issues/1303) it is stated that this seems to have been temporary and currently no work is being planned (26/11). I couldn't find any more information, so currently the future of the library seems to be unsure.

What should we do about that? Wait and see how they are going to continue the project or switch to an alternative?

**Comment by PhilippWendler** [comment]
Yes, given that we do not have a concrete need for replacing it, we should just wait.

**Comment by vinayak-00017** [comment]
it's a very old conversation, a lot might have happened. So what is the state of this issue? Are we going to replace the library? Is there any specific library that is being considered?

**Comment by PhilippWendler** [comment]
This is a good question. On our side nothing has happened. It seems that there have been some [releases](https://www.npmjs.com/package/react-vis?activeTab=versions) for react-vis, but they are not listed on GitHub and I cannot find release notes, which makes them dubious. It is questionable whether react-vis is still under good maintainership. Also, support for React 18 is [questionable](https://github.com/uber/react-vis/issues/1485).

I guess it depends whether we find a good replacement. I do not know of any alternative libraries right now.

**Comment by JawHawk** [comment]
I think [recharts](https://github.com/recharts/recharts) will be the best preferable replacement. I have used it many times and it has most of the features.

**Comment by PhilippWendler** [comment]
Seems plausible, at least as a first glance it looks good. But maybe there are other libraries as well, would be good to get some overview.
